### ISSUE TITLE
Spike: Research typecasting / typechecking existing API endpoints

### ISSUE THREAD

**Issue by jonathansadowski** [woothemes/woocommerce#30503]
One of the problems that often occurs for consumers of the API is that the values returned aren't of the type that that might be expecting. This may be happening because a filter along the way is changing the value returned to another type.

The desired outcome for this spike is a proposal for type-casting or type-checking existing endpoints.

Considerations:
* Would this only happen in the presence of a new parameter (to maintain backwards compat?)
* Should we cast the value as well?
* How should the warnings look?

**Comment by jonathansadowski** [comment]
@claudiosanches provided this documentation: https://developer.wordpress.org/reference/functions/rest_validate_value_from_schema/

**Comment by coreymckrill** [comment]
@barryhughes seems like #47918 would address this when it's complete... wdyt?

**Comment by barryhughes** [comment]
Yep, good find. I think it would pretty much cover the goals defined here.
